exonum-java-binding-common/src/main/java/com/exonum/binding/common/message/BinaryTransactionMessage.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.exonum.binding.common.message;
+
+import static com.exonum.binding.common.hash.Hashing.defaultHashFunction;
+
+import com.exonum.binding.common.crypto.PublicKey;
+import com.exonum.binding.common.hash.HashCode;
+import com.google.common.base.Objects;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/**
+ * Binary implementation of the {@link TransactionMessage} class.
+ */
+public final class BinaryTransactionMessage implements TransactionMessage {
+  private final ByteBuffer rawTransaction;
+
+  private BinaryTransactionMessage(ByteBuffer rawTransaction) {
+    this.rawTransaction = rawTransaction.duplicate().order(ByteOrder.LITTLE_ENDIAN);
+    this.rawTransaction.position(0);
```

Comment:
It's not OK to do with a ByteBuffer *unless* you make it a precondition
that a buffer has zero position **and** enforce that. But that limits the applicability
of this class.

Therefore, I'd recommend an alternative that is to slice a buffer:
```java
this.rawTransaction = rawTransaction.slice().order(…);
```